core/src/params.rs

```diff
@@ -0,0 +1,248 @@
+			} else {
+				self.bytes.push(self.end as u8);
+			}
+
+			// Safety: This is safe because JSON does not emit invalid UTF-8.
+			unsafe { String::from_utf8_unchecked(self.bytes) }
+		}
+	}
+}
+
+/// Parameter builder that serializes named value parameters to a JSON compatible string.
+/// This is the equivalent of a JSON Map object `{ key: value }`.
+///
+/// # Examples
+///
+/// ```rust
+///
+/// use jsonrpsee_core::params::ObjectParamsBuilder;
+///
+/// let mut builder = ObjectParamsBuilder::new();
+/// builder.insert("param1", 1);
+/// builder.insert("param2", "abc");
+/// let params = builder.build();
+///
+/// // Use RPC parameters...
+/// ```
+#[derive(Debug)]
+pub struct ObjectParamsBuilder(params_builder::ParamsBuilder);
+
+impl ObjectParamsBuilder {
+	/// Construct a new [`ObjectParamsBuilder`].
+	pub fn new() -> Self {
+		Self::default()
+	}
+
+	/// Insert a named value (key, value) pair into the builder.
+	/// The _name_ and _value_ are delimited by the `:` token.
+	pub fn insert<P: Serialize>(&mut self, name: &str, value: P) -> Result<(), serde_json::Error> {
+		self.0.insert_named(name, value)
+	}
+
+	/// Finish the building process and return a JSON compatible string.
+	pub fn build(self) -> ObjectParams {
+		ObjectParams(self.0.build())
+	}
+}
+
+impl Default for ObjectParamsBuilder {
+	fn default() -> Self {
+		Self(params_builder::ParamsBuilder::named())
+	}
+}
+
+/// Object RPC parameters stored as a JSON Map object `{ key: value }`.
+#[derive(Clone, Debug)]
+pub struct ObjectParams(String);
+
+impl ToRpcParams for ObjectParams {
+	fn to_rpc_params(self) -> Result<Option<Box<RawValue>>, Error> {
+		RawValue::from_string(self.0).map(Some).map_err(Error::ParseError)
+	}
+}
+
+/// Parameter builder that serializes plain value parameters to a JSON compatible string.
+/// This is the equivalent of a JSON Array object `[ value0, value1, .., valueN ]`.
+///
+/// # Examples
+///
+/// ```rust
+///
+/// use jsonrpsee_core::params::ArrayParamsBuilder;
+///
+/// let mut builder = ArrayParamsBuilder::new();
+/// builder.insert("param1");
+/// builder.insert(1);
+/// let params = builder.build();
+///
+/// // Use RPC parameters...
+/// ```
+#[derive(Debug)]
+pub struct ArrayParamsBuilder(params_builder::ParamsBuilder);
+
+impl ArrayParamsBuilder {
+	/// Construct a new [`ArrayParamsBuilder`].
+	pub fn new() -> Self {
+		Self::default()
+	}
+
+	/// Insert a plain value into the builder.
+	pub fn insert<P: Serialize>(&mut self, value: P) -> Result<(), serde_json::Error> {
+		self.0.insert(value)
+	}
+
+	/// Finish the building process and return a JSON compatible string.
+	pub fn build(self) -> ArrayParams {
+		ArrayParams(self.0.build())
+	}
+}
+
+impl Default for ArrayParamsBuilder {
+	fn default() -> Self {
+		Self(params_builder::ParamsBuilder::positional())
+	}
+}
+
+/// Array RPC parameters stored as a JSON Array object `[ value0, value1, .., valueN ]`.
+#[derive(Clone, Debug)]
+pub struct ArrayParams(String);
```

Comment:
Just a thought; do we need `ArrayParamsBuilder` + `ArrayParams` and same for objectparams?

We could just have `ArrayParams` have the functions that the current builder does, and also impl `ToRpcParams` (since that takes ownership of self anyway); would be simpler that way?